ansible_collections/arista/avd/roles/eos_designs/python_modules/connected_endpoints/port_channel_interfaces.py

```diff
@@ -0,0 +1,172 @@
+from __future__ import annotations
+
+import re
+from collections import ChainMap
+from functools import cached_property
+
+from ansible_collections.arista.avd.plugins.filter.esi_management import generate_esi, generate_lacp_id, generate_route_target
+from ansible_collections.arista.avd.plugins.filter.range_expand import range_expand
+from ansible_collections.arista.avd.plugins.plugin_utils.strip_empties import strip_null_from_data
+from ansible_collections.arista.avd.plugins.plugin_utils.utils import get
+
+from .utils import UtilsMixin
+
+
+class PortChannelInterfacesMixin(UtilsMixin):
+    """
+    Mixin Class used to generate structured config for one key.
+    Class should only be used as Mixin to a AvdStructuredConfig class
+    """
+
+    @cached_property
+    def port_channel_interfaces(self) -> dict | None:
+        """
+        Return structured config for ethernet_interfaces
+        """
+        port_channel_interfaces = {}
+        for connected_endpoint in self._filtered_connected_endpoints:
+            for adapter in connected_endpoint["adapters"]:
+                if get(adapter, "port_channel.mode") is None:
+                    continue
+
+                default_channel_group_id = int("".join(re.findall(r"\d", adapter["switch_ports"][0])))
+                channel_group_id = get(adapter, "port_channel.channel_id", default=default_channel_group_id)
+
+                port_channel_interface_name = f"Port-Channel{channel_group_id}"
+                port_channel_interfaces[port_channel_interface_name] = self._get_port_channel_interface_cfg(adapter, channel_group_id, connected_endpoint)
+                if (subinterfaces := get(adapter, "port_channel.subinterfaces")) is None:
+                    continue
+
+                for subinterface in subinterfaces:
+                    if "number" not in subinterface:
+                        continue
+
+                    port_channel_subinterface_name = f"Port-Channel{channel_group_id}.{subinterface['number']}"
+                    port_channel_interfaces[port_channel_subinterface_name] = self._get_port_channel_subinterface_cfg(subinterface, adapter, channel_group_id)
+
+        for network_port in self._filtered_network_ports:
+            if get(network_port, "port_channel.mode") is None:
+                continue
+
+            connected_endpoint = {
+                "name": network_port.get("description"),
+                "type": "network_port",
+            }
+            for ethernet_interface_name in range_expand(network_port["switch_ports"]):
+                # Override switches and switch_ports to only render for a single interface
+                # The blank extra switch is only inserted to work around port_channel validations
+                # This also means that port-channels defined with network_ports data model will be single-port per switch.
+                # Caveat: "short_esi: auto" and "designated_forwarder_algorithm: auto" will not work correctly.
```

Comment:
Let's create an issue to fix it for 4.0?

Reply:
I don't think there is a real good solution here, unless we introduce a some new knobs.